apps/web/client/src/components/store/editor/sandbox/index.ts

```
@@ -775,6 +792,169 @@ export class SandboxManager {
+                console.error('Screenshot: Upload failed');
+                return false;
+            }
+
+            const dbPreviewImg = fromPreviewImg({
+                type: 'storage',
+                storagePath: { bucket: STORAGE_BUCKETS.PREVIEW_IMAGES, path: uploadRes.path },
+            });
+
+            // Update project through the editor engine's API access
+            const { api } = await import('@/trpc/react');
+            const updateProject = api.project.update.useMutation();
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)